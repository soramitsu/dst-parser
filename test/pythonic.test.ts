import { parseFile } from './test_util'
import { describe, expect, it } from 'vitest'

describe('Python syntax tests', () => {
    it('Code section', () => {
        let parsed = parseFile('code_section.py')
        expect(parsed).toMatchSnapshot()
    })
    
    it('Indentation levels', () => {
        let parsed = parseFile('code_section_2.py')
        expect(parsed).toMatchSnapshot()
    })
    
    it('Test for Python-style comments', () => {
        let parsed = parseFile('normal_py_style.py')
        expect(parsed).toMatchSnapshot()
    })
    
    it('Nested fragment escape', () => {
        let parsed = parseFile('nested_fragment_escape.py')
        expect(parsed).toMatchSnapshot()
    })
    
    it('A nested escape with a fragment placed inside the escape text', () => {
        let parsed = parseFile('nested_escape_test.py')
        expect(parsed).toMatchSnapshot()
    })
})