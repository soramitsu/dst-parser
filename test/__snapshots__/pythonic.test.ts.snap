// Vitest Snapshot v1

exports[`Python syntax tests > A nested escape with a fragment placed inside the escape text 1`] = `
{
  "TestA": "This part is displayed",
}
`;

exports[`Python syntax tests > Code section 1`] = `
{
  "PythonStyleFragment": "print(\\"A normal Python code fragment\\")",
}
`;

exports[`Python syntax tests > Indentation levels 1`] = `
{
  "PythonStyleFragmentA": "def main():
    print(\\"A normal Pythonic fragment\\")",
  "PythonStyleFragmentB": "self.a = 0
self.b = 1",
}
`;

exports[`Python syntax tests > Nested fragment escape 1`] = `
{
  "TestA": "This part
is displayed",
  "TestB": "is displayed",
}
`;

exports[`Python syntax tests > Test for Python-style comments 1`] = `
{
  "PythonicFragment": "print('TEST')",
}
`;
